stable-2.4.0.md

```diff
@@ -0,0 +1,188 @@
+  * Added the `--linkerd-cni-enabled` flag to the `install` subcommands so that
+    `NET_ADMIN` capability is omitted from the CNI-enabled control plane's PSP
+  * Updated `linkerd check` to validate the caller can create
+    `PodSecurityPolicy` resources
+  * Added a check to `linkerd install` to prevent installing multiple control
+    planes into different namespaces
+  * Added support for passing a URL directly to `linkerd inject` (thanks
+    @Pothulapati!)
+  * Added the `--all-namespaces` flag to `linkerd edges`
+  * Added more descriptive output to the `linkerd check` output for control
+    plane ReplicaSet readiness
+  * **Breaking change** Renamed `config.linkerd.io/debug` annotation to
+    `config.linkerd.io/enable-debug-sidecar`, to match the
+    `--enable-debug-sidecar` CLI flag that sets it
+  * Fixed a bug in `linkerd edges` that caused incorrect identities to be
+    displayed when requests were sent from two or more namespaces
+* Controller
+  * Added Go pprof HTTP endpoints to all control plane components' admin servers
+    to better assist debugging efforts
+  * Fixed bug in the proxy injector, where sporadically the pod workload owner
+    wasn't properly determined, which would result in erroneous stats
+  * Added support for a new `config.linkerd.io/disable-identity` annotation to
+    opt out of identity for a specific pod
+  * Fixed pod creation failure when a `ResourceQuota` exists by adding a default
+    resource spec for the proxy-init init container
+  * Fixed control plane components failing on startup when the Kubernetes API
+    returns an `ErrGroupDiscoveryFailed`
+  * Added Controller Component Labels to the webhook config resources (thanks,
+    @Pothulapati!)
+  * Moved the tap service into its own pod
+  * **New** Control plane installations now generate a self-signed certificate
+    and private key pair for each webhook, to prepare for future work to make
+    the proxy injector and service profile validator HA
+  * Added a debug container annotation, allowing the `--enable-debug-sidecar`
+    flag to work when auto-injecting Linkerd proxies
+  * Added multiple replicas for the `proxy-injector` and `sp-validator`
+    controllers when run in high availability mode (thanks to @Pothulapati!)
+  * Defined least privilege default security context values for the proxy
+    container so that auto-injection does not fail on (thanks @codeman9!)
+  * Default the webhook failure policy to `Fail` in order to account for
+    unexpected errors during auto-inject; this ensures uninjected applications
+    are not deployed
+  * Introduced control plane's PSP and RBAC resources into Helm templates; these
+    policies are only in effect if the PSP admission controller is enabled
+  * Removed `UPDATE` operation from proxy-injector webhook because pod mutations
+    are disallowed during update operations
+  * Default the mutating and validating webhook configurations `sideEffects`
```

Comment:
Adding the `sideEffects` flag by default, breaks linkerd installations running on cluster versions < 1.12, `--omit-webhook-side-effects` can be used to install in those cases. 
I am guessing, A mention regarding this would be useful for users. WDYT?

Reply:
I am going to make a mention of the k8s pre-1.12 compatibility issue in the upgrade notes. Not sure if we need to mention it here. If anything, we should mention that k8s 1.12 is now the official minimum supported version.